src/framework-ondata.js

```diff
@@ -312,13 +312,18 @@ function handleOutput( port, lastLm, state ) {
 
         // Got any edges out of this port? 
         if (port.listAttached().length > 0) {
+            var nodeInstance = port.nodeInstance;
             logger.debug('sending state', {
                 state: util.inspect(state,{depth:1}),
-                nodeInstance: port.nodeInstance
+                nodeInstance: nodeInstance
             });
 
             port.send( state );
             port.disconnect();
+
+            if (nodeInstance.isTransient) { 
+                 nodeInstance.deleteAllVnis();
```

Comment:
This should only delete the current VNI.  I think I initially said it should delete all of them, but I realized later that it should only delete the current one, because they others may not have fired yet.  Sorry about that!